Sources/web3swift/Web3/Web3+Structures.swift

```diff
@@ -442,130 +236,94 @@ public struct Block: Decodable {
+        self.baseFeePerGas = try container.decodeHex(to: BigUInt.self, key: .baseFeePerGas)
 
-        guard let totalDifficulty = try decodeHexToBigUInt(container, key: .totalDifficulty) else {throw Web3Error.dataError}
-        self.totalDifficulty = totalDifficulty
+        self.timestamp = try container.decodeHex(to: Date.self, key: .timestamp)
 
-        guard let extraData = try decodeHexToData(container, key: .extraData) else {throw Web3Error.dataError}
-        self.extraData = extraData
-
-        guard let size = try decodeHexToBigUInt(container, key: .size) else {throw Web3Error.dataError}
-        self.size = size
-
-        guard let gasLimit = try decodeHexToBigUInt(container, key: .gasLimit) else {throw Web3Error.dataError}
-        self.gasLimit = gasLimit
-
-        guard let gasUsed = try decodeHexToBigUInt(container, key: .gasUsed) else {throw Web3Error.dataError}
-        self.gasUsed = gasUsed
-
-        let timestampString = try container.decode(String.self, forKey: .timestamp).stripHexPrefix()
-        guard let timestampInt = UInt64(timestampString, radix: 16) else {throw Web3Error.dataError}
-        let timestamp = Date(timeIntervalSince1970: TimeInterval(timestampInt))
-        self.timestamp = timestamp
-
-        let transactions = try container.decode([TransactionInBlock].self, forKey: .transactions)
-        self.transactions = transactions
+        self.transactions = try container.decode([TransactionInBlock].self, forKey: .transactions)
 
         let unclesStrings = try container.decode([String].self, forKey: .uncles)
-        var uncles = [Data]()
-        for str in unclesStrings {
-            guard let d = Data.fromHex(str) else {throw Web3Error.dataError}
-            uncles.append(d)
+        self.uncles = try unclesStrings.map {
+            guard let data = Data.fromHex($0) else { throw Web3Error.dataError }
+            return data
         }
-        self.uncles = uncles
     }
 }
 
-public struct EventParserResult: EventParserResultProtocol {
+public struct EventParserResult:EventParserResultProtocol {
     public var eventName: String
     public var transactionReceipt: TransactionReceipt?
     public var contractAddress: EthereumAddress
-    public var decodedResult: [String: Any]
-    public var eventLog: EventLog?
-
-    public init (eventName: String, transactionReceipt: TransactionReceipt?, contractAddress: EthereumAddress, decodedResult: [String: Any]) {
-        self.eventName = eventName
-        self.transactionReceipt = transactionReceipt
-        self.contractAddress = contractAddress
-        self.decodedResult = decodedResult
-        self.eventLog = nil
-    }
+    public var decodedResult: [String:Any]
+    public var eventLog: EventLog? = nil
 }
 
 public struct TransactionSendingResult {
     public var transaction: EthereumTransaction
     public var hash: String
 }
 
-public struct TxPoolStatus: Decodable {
+public struct TxPoolStatus : Decodable {
```

Comment:
Redundant space here in `TxPoolStatus :`.

Reply:
This is inherited from PR #510 please provide feedback there. If there are any lint issues lingering after 510 is merged into develop, but before this one is, I will make the adjustments then. For the time being I'm trying not to alter the same files as he is to make merges easier.